### PR TITLE
Storybook: Update to latest 5.3

### PR DESCRIPTION
**Pull request by ItsJonQ**
Screen Shot 2020-01-13 at 11 09 22 AM

This update bumps Storybook to the latest 5.3.2 release. Some of the core `.js`
files have been renamed and refactored according to their [5.2.x -> 5.3
migration guides](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-52x-to-53x).

This update also start some Storybook theming, adding the WordPress branding
logo on the top left.

### #branding

The WP logo is being served from a URL. Originally, I tried to serve the image locally. However, I don't think it's working correctly with Storybook.

https://github.com/storybookjs/storybook/issues/5878
https://github.com/storybookjs/storybook/issues/6521

Maybe able to resolve this with Webpack work-arounds?

## How has this been tested?
Run storybook locally via `npm run storybook:dev`. Also did a local build via `npm run storybook:build`. I tested the local `dist` build of storybook separately to ensure that it will work when built/deployed.

## Types of changes
* Bump Storybook dependencies from `5.2.4` -> `5.3.2`
* Adjusted main Storybook files according to their migration guides
* Added WP logo 🙌 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+	addons: [
+		'@storybook/addon-docs/react/preset',
```
Comment (epiqueras) [inline storybook/main.js]: They use `name: '@storybook/addon-docs',` in the migration guide.

File: storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+	addons: [
```
Comment (epiqueras) [inline storybook/main.js]: Shouldn't the stories RegExp be here now?

File: storybook/preview.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ import './style.scss';
 
 addDecorator( withA11y );
 addDecorator( withKnobs );
+
 configure(
```
Comment (epiqueras) [inline storybook/preview.js]: This should be part of `main.js` now, right?

File: storybook/main.js (hunk 1 above)
Comment (epiqueras) [reply storybook/main.js]: >Addon-docs configuration gets simpler in 5.3. In 5.2, each framework had its own preset, e.g. @storybook/addon-docs/react/preset. Starting in 5.3, everybody should use @storybook/addon-docs/preset.

**Review by epiqueras — CHANGES_REQUESTED** [review]
We still need to deal with:

>- We've deprecated the ability to specify the hierarchy separators (how you control the grouping of story kinds in the sidebar). From Storybook 6.0 we will have a single separator /, which cannot be configured.
>- In 5.3, the CSF loader decouples the story's name/id, which means that displayName is no longer necessary. Unfortunately, this is a breaking change for any code that uses the story name field. Storyshots relies on story name, and the appropriate migration is to simply update your snapshots. Apologies for the inconvenience!

File: storybook/main.js (hunk 1 above)
Comment (ItsJonQ) [reply storybook/main.js]: @epiqueras Thank you! 

I'll make those updates. I'm also not 100% clear if the add-ons need to be Objects to work correctly. It seemed to work as strings (at least for me). I imagined it worked similarly to how Babel's config worked.

File: storybook/preview.js (hunk 3 above)
Comment (ItsJonQ) [reply storybook/preview.js]: That's my guess as well. However, I wasn't sure how to do it given the conditional `require` we currently have. I'll give it a shot!

File: storybook/main.js (hunk 1 above)
Comment (epiqueras) [reply storybook/main.js]: They can be strings if you don't need extra configuration.

**Comment by ItsJonQ** [comment]
@epiqueras I just pushed the updates to the `main.js` config. I've also updated the separators (previously `|`) to the new Storybook standardized `/`.

**Comment by epiqueras** [comment]
Thanks, we still need to update the snapshots.

**Comment by ItsJonQ** [comment]
@epiqueras Oh! Thanks for the reminder <3
I'm working on that now. Jest is throwing some DOM related errors. Not sure what's causing that. 

```
  ● Storyshots › Components/ClipboardButton › Default

    TypeError: First argument must be a String, HTMLElement, HTMLCollection, or NodeList

      28 | 		const container = this.containerRef.current;
      29 |
    > 30 | 		this.clipboard = new Clipboard( container.firstChild,	{
         | 		                 ^
      31 | 			text: getText,
      32 | 			container,
      33 | 		} );
```

Taking a look at the test file, it looks like there may be something to help with that:

```
			if ( story.kind === 'Components|ClipboardButton' ) {
				currentElement.appendChild(
					document.createElement( 'button' )
				);
			}
``

Will keep investigating

**Comment by epiqueras** [comment]
You need to mock the refs.

**Comment by ItsJonQ** [comment]
Rebasing with latest `master`

**Comment by gziolo** [comment]
https://travis-ci.com/WordPress/gutenberg/jobs/275074009#L398

It looks like types checking errors for some unknown reasons.

**Comment by ItsJonQ** [comment]
@gziolo Interesting 🤔. I'm not sure what's up with that. I'll poke at it to see if I can debug.

Thank you!!

**Comment by ItsJonQ** [comment]
@gziolo Halloo!! I think I fixed it 🙏 . I installed the latest `@types/jest` as a devDependency. It looks like the project didn't have it. I think the type definitions were failing because an older version of `@types/jest` may have been a dependency of another dependency.

I based this fix off of this GH issue:
https://github.com/testing-library/jest-dom/issues/152

**Review by epiqueras — APPROVED** [review]

**Comment by ItsJonQ** [comment]
Thanks for the review @epiqueras !! Will merge 💪

**Pull request merged**